### ISSUE TITLE
Fatal signal 6 (SIGABRT), code -1 (SI_QUEUE) in tid 8244 (1.raster), pid 8207 (ample.test_isar)

### ISSUE THREAD

**Issue by AlexisL61** [isar/isar#903]
App crash with the error : Fatal signal 6 (SIGABRT), code -1 (SI_QUEUE) in tid 8244 (1.raster), pid 8207 (ample.test_isar)
Full error : 
```
I/scudo   ( 8207): Scudo ERROR: corrupted chunk header at address 0x797e6b5ba9b0
F/libc    ( 8207): Fatal signal 6 (SIGABRT), code -1 (SI_QUEUE) in tid 8244 (1.raster), pid 8207 (ample.test_isar)
*** *** *** *** *** *** *** *** *** *** *** *** *** *** *** ***
Build fingerprint: 'google/sdk_gphone64_x86_64/emu64xa:13/TPB4.220624.004/8808248:userdebug/dev-keys'
Revision: '0'
ABI: 'x86_64'
Timestamp: 2022-11-16 16:12:18.877520973+0100
Process uptime: 16s
Cmdline: com.example.test_isar
pid: 8207, tid: 8244, name: 1.raster  >>> com.example.test_isar <<<
uid: 10158
signal 6 (SIGABRT), code -1 (SI_QUEUE), fault addr --------
Abort message: 'Scudo ERROR: corrupted chunk header at address 0x797e6b5ba9b0'
    rax 0000000000000000  rbx 000000000000200f  rcx 00007980f30d068f  rdx 0000000000000006
    r8  0000000000000000  r9  0000000000000000  r10 0000797de89f99c0  r11 0000000000000203
    r12 0000797de625f160  r13 0000000000000023  r14 0000797de89f99b8  r15 0000000000002034
    rdi 000000000000200f  rsi 0000000000002034
    rbp 0000000000000000  rsp 0000797de89f99b0  rip 00007980f30d068f
backtrace:
      #00 pc 000000000005f68f  /apex/com.android.runtime/lib64/bionic/libc.so (abort+191) (BuildId: 57def992cb1772e13608c8efcafd893b)
      #01 pc 0000000000048d65  /apex/com.android.runtime/lib64/bionic/libc.so (scudo::die()+5) (BuildId: 57def992cb1772e13608c8efcafd893b)
      #02 pc 0000000000049358  /apex/com.android.runtime/lib64/bionic/libc.so (scudo::ScopedErrorReport::~ScopedErrorReport()+24) (BuildId: 57def992cb1772e13608c8efcafd893b)
      #03 pc 0000000000049442  /apex/com.android.runtime/lib64/bionic/libc.so (scudo::reportHeaderCorruption(void*)+66) (BuildId: 57def992cb1772e13608c8efcafd893b)
      #04 pc 000000000004ac69  /apex/com.android.runtime/lib64/bionic/libc.so (scudo::Allocator<scudo::AndroidConfig, &(scudo_malloc_postinit)>::deallocate(void*, scudo::Chunk::Origin, unsigned long, unsigned long)+313) (BuildId: 57def992cb1772e13608c8efcafd893b)
      #05 pc 0000000001c7f19b  /data/app/~~8PbF_tvuE9gBZdByqtu29A==/com.example.test_isar-TT25R09w9faWVSo1o-0ZbA==/lib/x86_64/libflutter.so (BuildId: b19d2b3189fb7c988d90df73622911a55f5aa88f)
      #06 pc 0000000001c7f791  /data/app/~~8PbF_tvuE9gBZdByqtu29A==/com.example.test_isar-TT25R09w9faWVSo1o-0ZbA==/lib/x86_64/libflutter.so (BuildId: b19d2b3189fb7c988d90df73622911a55f5aa88f)
      #07 pc 0000000001c7eff8  /data/app/~~8PbF_tvuE9gBZdByqtu29A==/com.example.test_isar-TT25R09w9faWVSo1o-0ZbA==/lib/x86_64/libflutter.so (BuildId: b19d2b3189fb7c988d90df73622911a55f5aa88f)
      #08 pc 0000000001c7e5af  /data/app/~~8PbF_tvuE9gBZdByqtu29A==/com.example.test_isar-TT25R09w9faWVSo1o-0ZbA==/lib/x86_64/libflutter.so (BuildId: b19d2b3189fb7c988d90df73622911a55f5aa88f)
      #09 pc 0000000001e88c95  /data/app/~~8PbF_tvuE9gBZdByqtu29A==/com.example.test_isar-TT25R09w9faWVSo1o-0ZbA==/lib/x86_64/libflutter.so (BuildId: b19d2b3189fb7c988d90df73622911a55f5aa88f)
      #10 pc 0000000001666f4b  /data/app/~~8PbF_tvuE9gBZdByqtu29A==/com.example.test_isar-TT25R09w9faWVSo1o-0ZbA==/lib/x86_64/libflutter.so (BuildId: b19d2b3189fb7c988d90df73622911a55f5aa88f)
      #11 pc 00000000019afbee  /data/app/~~8PbF_tvuE9gBZdByqtu29A==/com.example.test_isar-TT25R09w9faWVSo1o-0ZbA==/lib/x86_64/libflutter.so (BuildId: b19d2b3189fb7c988d90df73622911a55f5aa88f)
      #12 pc 00000000019af697  /data/app/~~8PbF_tvuE9gBZdByqtu29A==/com.example.test_isar-TT25R09w9faWVSo1o-0ZbA==/lib/x86_64/libflutter.so (BuildId: b19d2b3189fb7c988d90df73622911a55f5aa88f)
      #13 pc 00000000019b41bf  /data/app/~~8PbF_tvuE9gBZdByqtu29A==/com.example.test_isar-TT25R09w9faWVSo1o-0ZbA==/lib/x86_64/libflutter.so (BuildId: b19d2b3189fb7c988d90df73622911a55f5aa88f)
      #14 pc 00000000019af697  /data/app/~~8PbF_tvuE9gBZdByqtu29A==/com.example.test_isar-TT25R09w9faWVSo1o-0ZbA==/lib/x86_64/libflutter.so (BuildId: b19d2b3189fb7c988d90df73622911a55f5aa88f)
      #15 pc 00000000019b41bf  /data/app/~~8PbF_tvuE9gBZdByqtu29A==/com.example.test_isar-TT25R09w9faWVSo1o-0ZbA==/lib/x86_64/libflutter.so (BuildId: b19d2b3189fb7c988d90df73622911a55f5aa88f)
      #16 pc 00000000019af697  /data/app/~~8PbF_tvuE9gBZdByqtu29A==/com.example.test_isar-TT25R09w9faWVSo1o-0ZbA==/lib/x86_64/libflutter.so (BuildId: b19d2b3189fb7c988d90df73622911a55f5aa88f)
      #17 pc 00000000019b41bf  /data/app/~~8PbF_tvuE9gBZdByqtu29A==/com.example.test_isar-TT25R09w9faWVSo1o-0ZbA==/lib/x86_64/libflutter.so (BuildId: b19d2b3189fb7c988d90df73622911a55f5aa88f)
      #18 pc 00000000019af697  /data/app/~~8PbF_tvuE9gBZdByqtu29A==/com.example.test_isar-TT25R09w9faWVSo1o-0ZbA==/lib/x86_64/libflutter.so (BuildId: b19d2b3189fb7c988d90df73622911a55f5aa88f)
      #19 pc 00000000019b41bf  /data/app/~~8PbF_tvuE9gBZdByqtu29A==/com.example.test_isar-TT25R09w9faWVSo1o-0ZbA==/lib/x86_64/libflutter.so (BuildId: b19d2b3189fb7c988d90df73622911a55f5aa88f)
      #20 pc 00000000019af697  /data/app/~~8PbF_tvuE9gBZdByqtu29A==/com.example.test_isar-TT25R09w9faWVSo1o-0ZbA==/lib/x86_64/libflutter.so (BuildId: b19d2b3189fb7c988d90df73622911a55f5aa88f)
      #21 pc 00000000019b41bf  /data/app/~~8PbF_tvuE9gBZdByqtu29A==/com.example.test_isar-TT25R09w9faWVSo1o-0ZbA==/lib/x86_64/libflutter.so (BuildId: b19d2b3189fb7c988d90df73622911a55f5aa88f)
      #22 pc 00000000019af697  /data/app/~~8PbF_tvuE9gBZdByqtu29A==/com.example.test_isar-TT25R09w9faWVSo1o-0ZbA==/lib/x86_64/libflutter.so (BuildId: b19d2b3189fb7c988d90df73622911a55f5aa88f)
      #23 pc 00000000019ae3b3  /data/app/~~8PbF_tvuE9gBZdByqtu29A==/com.example.test_isar-TT25R09w9faWVSo1o-0ZbA==/lib/x86_64/libflutter.so (BuildId: b19d2b3189fb7c988d90df73622911a55f5aa88f)
      #24 pc 00000000019ae681  /data/app/~~8PbF_tvuE9gBZdByqtu29A==/com.example.test_isar-TT25R09w9faWVSo1o-0ZbA==/lib/x86_64/libflutter.so (BuildId: b19d2b3189fb7c988d90df73622911a55f5aa88f)
      #25 pc 00000000019af697  /data/app/~~8PbF_tvuE9gBZdByqtu29A==/com.example.test_isar-TT25R09w9faWVSo1o-0ZbA==/lib/x86_64/libflutter.so (BuildId: b19d2b3189fb7c988d90df73622911a55f5aa88f)
      #26 pc 00000000019b41bf  /data/app/~~8PbF_tvuE9gBZdByqtu29A==/com.example.test_isar-TT25R09w9faWVSo1o-0ZbA==/lib/x86_64/libflutter.so (BuildId: b19d2b3189fb7c988d90df73622911a55f5aa88f)
      #27 pc 00000000019af697  /data/app/~~8PbF_tvuE9gBZdByqtu29A==/com.example.test_isar-TT25R09w9faWVSo1o-0ZbA==/lib/x86_64/libflutter.so (BuildId: b19d2b3189fb7c988d90df73622911a55f5aa88f)
      #28 pc 00000000019b41bf  /data/app/~~8PbF_tvuE9gBZdByqtu29A==/com.example.test_isar-TT25R09w9faWVSo1o-0ZbA==/lib/x86_64/libflutter.so (BuildId: b19d2b3189fb7c988d90df73622911a55f5aa88f)
      #29 pc 00000000019af697  /data/app/~~8PbF_tvuE9gBZdByqtu29A==/com.example.test_isar-TT25R09w9faWVSo1o-0ZbA==/lib/x86_64/libflutter.so (BuildId: b19d2b3189fb7c988d90df73622911a55f5aa88f)
      #30 pc 00000000019b41bf  /data/app/~~8PbF_tvuE9gBZdByqtu29A==/com.example.test_isar-TT25R09w9faWVSo1o-0ZbA==/lib/x86_64/libflutter.so (BuildId: b19d2b3189fb7c988d90df73622911a55f5aa88f)
      #31 pc 00000000019af697  /data/app/~~8PbF_tvuE9gBZdByqtu29A==/com.example.test_isar-TT25R09w9faWVSo1o-0ZbA==/lib/x86_64/libflutter.so (BuildId: b19d2b3189fb7c988d90df73622911a55f5aa88f)
      #32 pc 00000000019b41bf  /data/app/~~8PbF_tvuE9gBZdByqtu29A==/com.example.test_isar-TT25R09w9faWVSo1o-0ZbA==/lib/x86_64/libflutter.so (BuildId: b19d2b3189fb7c988d90df73622911a55f5aa88f)
      #33 pc 00000000019af697  /data/app/~~8PbF_tvuE9gBZdByqtu29A==/com.example.test_isar-TT25R09w9faWVSo1o-0ZbA==/lib/x86_64/libflutter.so (BuildId: b19d2b3189fb7c988d90df73622911a55f5aa88f)
      #34 pc 00000000019b41bf  /data/app/~~8PbF_tvuE9gBZdByqtu29A==/com.example.test_isar-TT25R09w9faWVSo1o-0ZbA==/lib/x86_64/libflutter.so (BuildId: b19d2b3189fb7c988d90df73622911a55f5aa88f)
      #35 pc 00000000019af697  /data/app/~~8PbF_tvuE9gBZdByqtu29A==/com.example.test_isar-TT25R09w9faWVSo1o-0ZbA==/lib/x86_64/libflutter.so (BuildId: b19d2b3189fb7c988d90df73622911a55f5aa88f)
      #36 pc 00000000019b41bf  /data/app/~~8PbF_tvuE9gBZdByqtu29A==/com.example.test_isar-TT25R09w9faWVSo1o-0ZbA==/lib/x86_64/libflutter.so (BuildId: b19d2b3189fb7c988d90df73622911a55f5aa88f)
      #37 pc 00000000019af697  /data/app/~~8PbF_tvuE9gBZdByqtu29A==/com.example.test_isar-TT25R09w9faWVSo1o-0ZbA==/lib/x86_64/libflutter.so (BuildId: b19d2b3189fb7c988d90df73622911a55f5aa88f)
      #38 pc 00000000019b41bf  /data/app/~~8PbF_tvuE9gBZdByqtu29A==/com.example.test_isar-TT25R09w9faWVSo1o-0ZbA==/lib/x86_64/libflutter.so (BuildId: b19d2b3189fb7c988d90df73622911a55f5aa88f)
      #39 pc 00000000019af697  /data/app/~~8PbF_tvuE9gBZdByqtu29A==/com.example.test_isar-TT25R09w9faWVSo1o-0ZbA==/lib/x86_64/libflutter.so (BuildId: b19d2b3189fb7c988d90df73622911a55f5aa88f)
      #40 pc 00000000019b41bf  /data/app/~~8PbF_tvuE9gBZdByqtu29A==/com.example.test_isar-TT25R09w9faWVSo1o-0ZbA==/lib/x86_64/libflutter.so (BuildId: b19d2b3189fb7c988d90df73622911a55f5aa88f)
      #41 pc 00000000019af697  /data/app/~~8PbF_tvuE9gBZdByqtu29A==/com.example.test_isar-TT25R09w9faWVSo1o-0ZbA==/lib/x86_64/libflutter.so (BuildId: b19d2b3189fb7c988d90df73622911a55f5aa88f)
      #42 pc 00000000019b41bf  /data/app/~~8PbF_tvuE9gBZdByqtu29A==/com.example.test_isar-TT25R09w9faWVSo1o-0ZbA==/lib/x86_64/libflutter.so (BuildId: b19d2b3189fb7c988d90df73622911a55f5aa88f)
      #43 pc 00000000019af697  /data/app/~~8PbF_tvuE9gBZdByqtu29A==/com.example.test_isar-TT25R09w9faWVSo1o-0ZbA==/lib/x86_64/libflutter.so (BuildId: b19d2b3189fb7c988d90df73622911a55f5aa88f)
      #44 pc 00000000019b41bf  /data/app/~~8PbF_tvuE9gBZdByqtu29A==/com.example.test_isar-TT25R09w9faWVSo1o-0ZbA==/lib/x86_64/libflutter.so (BuildId: b19d2b3189fb7c988d90df73622911a55f5aa88f)
      #45 pc 00000000019af697  /data/app/~~8PbF_tvuE9gBZdByqtu29A==/com.example.test_isar-TT25R09w9faWVSo1o-0ZbA==/lib/x86_64/libflutter.so (BuildId: b19d2b3189fb7c988d90df73622911a55f5aa88f)
      #46 pc 00000000019b2249  /data/app/~~8PbF_tvuE9gBZdByqtu29A==/com.example.test_isar-TT25R09w9faWVSo1o-0ZbA==/lib/x86_64/libflutter.so (BuildId: b19d2b3189fb7c988d90df73622911a55f5aa88f)
      #47 pc 00000000019ab6a2  /data/app/~~8PbF_tvuE9gBZdByqtu29A==/com.example.test_isar-TT25R09w9faWVSo1o-0ZbA==/lib/x86_64/libflutter.so (BuildId: b19d2b3189fb7c988d90df73622911a55f5aa88f)
      #48 pc 00000000019cc93a  /data/app/~~8PbF_tvuE9gBZdByqtu29A==/com.example.test_isar-TT25R09w9faWVSo1o-0ZbA==/lib/x86_64/libflutter.so (BuildId: b19d2b3189fb7c988d90df73622911a55f5aa88f)
      #49 pc 00000000019cc084  /data/app/~~8PbF_tvuE9gBZdByqtu29A==/com.example.test_isar-TT25R09w9faWVSo1o-0ZbA==/lib/x86_64/libflutter.so (BuildId: b19d2b3189fb7c988d90df73622911a55f5aa88f)
      #50 pc 00000000019ccfba  /data/app/~~8PbF_tvuE9gBZdByqtu29A==/com.example.test_isar-TT25R09w9faWVSo1o-0ZbA==/lib/x86_64/libflutter.so (BuildId: b19d2b3189fb7c988d90df73622911a55f5aa88f)
      #51 pc 00000000019cb823  /data/app/~~8PbF_tvuE9gBZdByqtu29A==/com.example.test_isar-TT25R09w9faWVSo1o-0ZbA==/lib/x86_64/libflutter.so (BuildId: b19d2b3189fb7c988d90df73622911a55f5aa88f)
      #52 pc 00000000019cb521  /data/app/~~8PbF_tvuE9gBZdByqtu29A==/com.example.test_isar-TT25R09w9faWVSo1o-0ZbA==/lib/x86_64/libflutter.so (BuildId: b19d2b3189fb7c988d90df73622911a55f5aa88f)
      #53 pc 00000000019d9582  /data/app/~~8PbF_tvuE9gBZdByqtu29A==/com.example.test_isar-TT25R09w9faWVSo1o-0ZbA==/lib/x86_64/libflutter.so (BuildId: b19d2b3189fb7c988d90df73622911a55f5aa88f)
      #54 pc 0000000001662e62  /data/app/~~8PbF_tvuE9gBZdByqtu29A==/com.example.test_isar-TT25R09w9faWVSo1o-0ZbA==/lib/x86_64/libflutter.so (BuildId: b19d2b3189fb7c988d90df73622911a55f5aa88f)
      #55 pc 000000000166825a  /data/app/~~8PbF_tvuE9gBZdByqtu29A==/com.example.test_isar-TT25R09w9faWVSo1o-0ZbA==/lib/x86_64/libflutter.so (BuildId: b19d2b3189fb7c988d90df73622911a55f5aa88f)
      #56 pc 000000000001076e  /system/lib64/libutils.so (android::Looper::pollOnce(int, int*, int*, void**)+894) (BuildId: bf3f66fd6f5b7c78306a1c8270331f22)
      #57 pc 00000000000192ff  /system/lib64/libandroid.so (ALooper_pollOnce+95) (BuildId: 85f684113baf73620c89a4da5255153d)
      #58 pc 00000000016681fd  /data/app/~~8PbF_tvuE9gBZdByqtu29A==/com.example.test_isar-TT25R09w9faWVSo1o-0ZbA==/lib/x86_64/libflutter.so (BuildId: b19d2b3189fb7c988d90df73622911a55f5aa88f)
      #59 pc 0000000001662dce  /data/app/~~8PbF_tvuE9gBZdByqtu29A==/com.example.test_isar-TT25R09w9faWVSo1o-0ZbA==/lib/x86_64/libflutter.so (BuildId: b19d2b3189fb7c988d90df73622911a55f5aa88f)
      #60 pc 0000000001666954  /data/app/~~8PbF_tvuE9gBZdByqtu29A==/com.example.test_isar-TT25R09w9faWVSo1o-0ZbA==/lib/x86_64/libflutter.so (BuildId: b19d2b3189fb7c988d90df73622911a55f5aa88f)
      #61 pc 00000000000cccba  /apex/com.android.runtime/lib64/bionic/libc.so (__pthread_start(void*)+58) (BuildId: 57def992cb1772e13608c8efcafd893b)
      #62 pc 0000000000060cc7  /apex/com.android.runtime/lib64/bionic/libc.so (__start_thread+55) (BuildId: 57def992cb1772e13608c8efcafd893b)
```

### Steps to Reproduce

Clone repo : https://github.com/AlexisL61/test_isar
flutter run
Restart the app by doing another flutter run or hot restart
Try moving in the list of session
The app crash

[isar.webm](https://user-images.githubusercontent.com/30233189/202224137-726e0091-fb66-4647-8599-1f9b27f375a2.webm)

### Details

 - Platforms: x86 Android Emulator, iPad mini 4, ARM android device API 23
 - Flutter version: 3.3.7
 - Isar version: 3.0.4

---

 - [x] I searched for similar issues already
 - [x] I filled the details section with the exact device model and version
 - [x] I am able to provide a reproducible example

**Comment by leisim** [comment]
@AlexisL61 You are my hero today! This example was incredibly helpful and after hours of searching I found the issue! Thanks so much for your effort putting this together. I'll release a fix first thing tomorrow.

**Comment by leisim** [comment]
@all-contributors please add @AlexisL61 for bug report

**Comment by leisim** [comment]
Fixed in 3.0.5 🙂

**Comment by kpradeepkumarreddy** [comment]
What is the cause for this Bug ?
